src/condor_utils/file_transfer.cpp

```diff
@@ -2023,6 +2064,41 @@ FileTransfer::ReadTransferPipeMsg()
 	return false;
 }
 
+bool
+FileTransfer::SendPluginOutputAd( const ClassAd & plugin_output_ad ) {
+	// Do we have a pipe to which to write?
+	if( TransferPipe[1] == -1 ) { return false; }
+
+	// Write the command.
+	char cmd = PLUGIN_OUTPUT_AD;
+	int n = daemonCore->Write_Pipe( TransferPipe[1], & cmd, sizeof(cmd) );
+	if( n != sizeof(cmd) ) { return false; }
+
+	// Serialize the ClassAd.
+	std::string plugin_output_ad_string;
+	classad::ClassAdUnParser caup;
+	caup.Unparse( plugin_output_ad_string, & plugin_output_ad );
+
+	// Write the size of the serialization.
+	int size_of_ad = plugin_output_ad_string.size();
+	n = daemonCore->Write_Pipe(
+		TransferPipe[1], & size_of_ad, sizeof(size_of_ad)
+	);
+	if( n != sizeof(size_of_ad) ) { return false; }
+
+	// Write the serialization.
+	n = daemonCore->Write_Pipe(
+		TransferPipe[1],
+		plugin_output_ad_string.c_str(),
+		plugin_output_ad_string.size()
+	);
+	// I'm also asserting that our ads aren't >= 2GB.
+	ASSERT( n == (int)plugin_output_ad_string.size() );
```

Comment:
are you sure you want to kill the process here?

Reply:
I'm open to suggestions, but I don't know what the forked child or for that matter the parent could or should do if it the child can't write the results to the pipe.

Reply:
dprintf and return false.  better to keep going than kill the process.  especially since this might not be a forked child.